src/commands/tags/TagFileSystem.ts

```diff
@@ -0,0 +1,95 @@
+import { getTagDiagnostics } from "./getTagDiagnostics";
+
+const insertKeyHere: string = localize('insertTagName', '<Insert tag name>');
+const insertValueHere: string = localize('insertTagValue', '<Insert tag value>');
+
+/**
+ * For now this file system only supports editing tags.
+ * However, the scheme was left generic so that it could support editing other stuff in this extension without needing to create a whole new file system
+ */
+export class TagFileSystem extends AzExtFileSystem<ResourceGroupTreeItem> {
+    public static scheme: string = 'azureResourceGroups';
+    public scheme: string = TagFileSystem.scheme;
+
+    public async statImpl(_context: IActionContext, node: ResourceGroupTreeItem): Promise<FileStat> {
+        const fileContent: string = this.getFileContentFromTags(node.data.tags);
+        return { type: FileType.File, ctime: node.cTime, mtime: node.mTime, size: Buffer.byteLength(fileContent) };
+    }
+
+    public async readFileImpl(_context: IActionContext, node: ResourceGroupTreeItem): Promise<Uint8Array> {
+        const fileContent: string = this.getFileContentFromTags(node.data.tags);
+        return Buffer.from(fileContent);
+    }
+
+    public async writeFileImpl(context: IActionContext, node: ResourceGroupTreeItem, content: Uint8Array, originalUri: Uri): Promise<void> {
+        const text: string = content.toString();
+
+        // tslint:disable-next-line: strict-boolean-expressions
+        const diagnostics: readonly Diagnostic[] = ext.diagnosticCollection.get(originalUri) || getTagDiagnostics(text);
+        if (diagnostics.length > 0) {
+            context.telemetry.measurements.tagDiagnosticsLength = diagnostics.length;
+
+            const showErrors: MessageItem = { title: localize('showErrors', 'Show Errors') };
+            const message: string = localize('errorsExist', 'Failed to upload tags for resource group "{0}".', node.name);
+            // don't wait
+            window.showErrorMessage(message, showErrors).then(async (result) => {
+                if (result === showErrors) {
+                    const openedUri: Uri | undefined = window.activeTextEditor?.document.uri;
+                    if (!openedUri || !this.areUrisEqual(originalUri, openedUri)) {
+                        await this.showTextDocument(node);
+                    }
+
+                    await commands.executeCommand('workbench.action.showErrorsWarnings');
+                }
+            });
+
+            context.errorHandling.suppressDisplay = true;
+            // This won't be displayed, but might as well track the first diagnostic for telemetry
+            throw new Error(diagnostics[0].message);
+        } else {
+            const message: string = localize('confirmTags', 'Are you sure you want to update tags for resource group "{0}"?', node.name);
+            const update: MessageItem = { title: localize('update', 'Update') };
+            await ext.ui.showWarningMessage(message, { modal: true }, update);
```

Comment:
No setting to disable the confirmation prompt?  I feel like most of our FileEditors do this-- maybe it could just be a property in the editor itself?

Reply:
If we could do something like this checkbox, I would be all for it (not sure if extensions can do this, though):
![Screen Shot 2020-06-04 at 1 43 02 PM](https://user-images.githubusercontent.com/11282622/83809158-d9db6600-a66a-11ea-9fb8-b0c9a603e9f7.png)

I think this is a little ugly/confusing:
![Screen Shot 2020-06-04 at 1 58 22 PM](https://user-images.githubusercontent.com/11282622/83809561-80c00200-a66b-11ea-9010-24ec0ff6f3db.png)

I would rather wait for users to complain about the confirmation dialog before I spend time making it ugly lol (at least for this scenario)

> maybe it could just be a property in the editor itself?

Not sure what you mean by this. One thing to note is that the warning message is in `TagFileSystem`, not `AzExtFileSystem` - so each extension can do whatever it wants for its specific scenario